dotcom-rendering/playwright/tests/parallel-4/ophan.e2e.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { isUndefined } from '@guardian/libs';
+import type { Page } from '@playwright/test';
+import { test } from '@playwright/test';
+import { cmpAcceptAll, cmpRejectAll, disableCMP } from '../../lib/cmp';
+import { loadPage } from '../../lib/load-page';
+
+const articleUrl =
+	'https://www.theguardian.com/politics/2019/oct/29/tories-restore-party-whip-to-10-mps-who-sought-to-block-no-deal-brexit';
+
+const frontUrl = 'https://www.theguardian.com/uk';
+
+const interceptOphanRequest = ({
+	page,
+	path,
+	searchParamMatcher,
+}: {
+	page: Page;
+	path: string;
+	searchParamMatcher: (searchParams: URLSearchParams) => boolean;
+}) => {
+	return page.waitForRequest((request) => {
+		const matchUrl = request
+			.url()
+			.startsWith(`https://ophan.theguardian.com/${path}`);
+		const searchParams = new URLSearchParams(request.url());
+		return matchUrl && searchParamMatcher(searchParams);
+	});
+};
+
+test.describe('Ophan requests', () => {
+	test('should make a view request on an article when consent is rejected', async ({
+		page,
+	}) => {
+		const ophanRequestPromise = interceptOphanRequest({
+			page,
+			path: 'img/1',
```

Comment:
Why `img/1` and `img/2`?

Reply:
Good question!

Looking at the network requests that Ophan makes:

- `img/1` is the single 'view' request that conveys the page URL being viewed.
- `img/2` are the multiple 'event' requests that convey individual pieces of data such as abtests, attentionMS, consent and performance metrics.

Its good to raise though. I will ask the Ophan team what their official naming is for these requests.